server/src/main/java/com/broadcom/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -55,19 +55,15 @@
  */
 @Slf4j
 public class CobolVisitor extends CobolParserBaseVisitor<Class> {
-  private static final String AREA_A_WARNING_MSG = "The following token must start in Area A: ";
-  private static final String AREA_B_WARNING_MSG = "The following token must start in Area B: ";
-  private static final String IDENTICAL_PROGRAM_MSG =
-      "Program-name must be identical to the program-name of the corresponding PROGRAM-ID paragraph: ";
   private static final String DECLARATIVE_SAME_MSG =
       "The following token cannot be on the same line as a DECLARATIVE token: ";
-  private static final String INVALID_DEF_MSG = "Invalid definition for: ";
-  private static final String MISSPELLED_WORD = "A misspelled word, maybe you want to put ";
-  private static final String PROGRAM_ID_ISSUE_MSG = "There is an issue with PROGRAM-ID paragraph";
 
-  @Getter private List<SyntaxError> errors = new ArrayList<>();
+  private List<SyntaxError> errors = new ArrayList<>();
 
   private SubContext<String> paragraphs = new NamedSubContext();
+  private Multimap<String, Locality> paragraphUsageLocalities = HashMultimap.create();
+  private Multimap<String, Locality> paragraphDefinitionLocalities = HashMultimap.create();
```

Comment:
Ok, will work for now. We will change the paragraph context to accept localities later on, altogether with the same late check for variables.

Reply:
Yep. Better to use `Locality` instead of `Location` internally and only for publishing to the client convert map values to `Location`.